app/controllers/tasks_controller.rb

```diff
@@ -13,35 +14,81 @@ class TasksController < ApplicationController
   end
 
   post '/tasks' do
-    task = Task.new(description: params[:description])
-
+    task = Task.new(user: current_user, description: params[:description])
     if task.save
-      redirect "/"
+      if is_json_request?
+        status 201
+        task.as_json.to_json
+      else
+        redirect "/"
+      end
     else
-      flash.now[:errors] = task.errors.full_messages.join("; ")
-      erb :"tasks/new.html"
+      if is_json_request?
+        status 400
+        {errors: task.errors.full_messages}.to_json
+      else
+        flash.now[:errors] = task.errors.full_messages.join("; ")
+        erb :"tasks/new.html"
+      end      
     end
   end
 
   get '/tasks/:id' do
-    task = Task.find(params[:id])
-    erb :"tasks/edit.html", locals: { task: task }
+    task = current_user.tasks.find_by_id(params[:id])
+    if is_json_request?
+      if task == nil
+        status 404
+        json({})
+      else
+        task.as_json.to_json
+      end
+    else
+      erb :"tasks/edit.html", locals: { task: task }
+    end
   end
 
   put '/tasks/:id' do
-    task = Task.find(params[:id])
-    task.description = params[:description]
-    if task.save
-      redirect "/"
+    task = current_user.tasks.find_by_id(params[:id])
+    if task == nil
```

Comment:
Just an FYI- Idiomatic ruby would use `if tack.nil?` instead of `if task == nil`